app/factory/Layer.js

```diff
@@ -0,0 +1,184 @@
+/**
+ * [extend description]
+ * @type {String}
+ */
+Ext.define('CpsiMapview.factory.Layer', {
+    alternateClassName: 'LayerFactory',
+    requires: [],
+
+    singleton: true,
+
+    /**
+     * Creates an OpenLayer layer object from a JSON config
+     *
+     * @param  {Object} layerConf  The configuration object
+     * @return {ol.layer.Base} OL layer object
+     */
+    createLayer: function(layerConf) {
+
+        var layerType = layerConf.layerType;
+        var mapLayer;
+
+        switch (layerType) {
+        case 'wms':
+            mapLayer = LayerFactory.createWms(layerConf);
+            break;
+        case 'wfs':
+            mapLayer = LayerFactory.createWFS(layerConf);
+            break;
+        case 'esrirest':
+            mapLayer = LayerFactory.createEsriRest(layerConf);
+            break;
+        case 'osm':
+            mapLayer = LayerFactory.createOsm(layerConf);
+            break;
+        case 'empty':
+            mapLayer = LayerFactory.createEmptyLayer(layerConf);
+            break;
+        case 'bing_aerial':
+            mapLayer = LayerFactory.createBing(layerConf, 'Aerial');
+            break;
+        case 'google_roadmap':
+            mapLayer = LayerFactory.createGoogle(layerConf, 'google.maps.MapTypeId.ROADMAP');
+            break;
+        case 'google_terrain':
+            mapLayer = LayerFactory.createGoogle(layerConf, 'google.maps.MapTypeId.TERRAIN');
+            break;
+        case 'google_hybrid':
+            mapLayer = LayerFactory.createGoogle(layerConf, 'google.maps.MapTypeId.HYBRID');
+            break;
+        case 'google_satellite':
+            mapLayer = LayerFactory.createGoogle(layerConf, 'google.maps.MapTypeId.SATELLITE');
+            break;
+        case 'nasa':
+            mapLayer = LayerFactory.createNASA(layerConf);
+            break;
+        case 'os':
+            mapLayer = LayerFactory.createOS(layerConf);
+            break;
+        case 'arcgiscache':
+            mapLayer = LayerFactory.createArcGISCache(layerConf);
+            break;
+        case 'arcgisrest':
+            mapLayer = LayerFactory.createArcGISRest(layerConf);
+            break;
+        case 'switchlayer':
+            mapLayer = LayerFactory.createSwitchLayer(layerConf);
+            break;
+        default:
+            Ext.log.warn('Layer type not known');
+            //do nothing, and return empty layer
+        }
+
+        return mapLayer;
+    },
+
+    createEmptyLayer: function(layerConf) {
+        Ext.log.info('Not implemented yet', layerConf);
+    },
```

Comment:
We should implement this, so more complex configs worl out of the box.

Reply:
They should because the addition of the layers checks if we have valid layer instances